core/src/main/scala/org/apache/spark/ui/storage/StorageTab.scala

```diff
@@ -59,7 +59,15 @@ class StorageListener(storageStatusListener: StorageStatusListener) extends Bloc
 
   override def onStageSubmitted(stageSubmitted: SparkListenerStageSubmitted): Unit = synchronized {
     val rddInfos = stageSubmitted.stageInfo.rddInfos
-    rddInfos.foreach { info => _rddInfoMap.getOrElseUpdate(info.id, info) }
+    rddInfos.foreach { info =>
```

Comment:
Hm I'm not so sure if it's safe to just replace it. We mutate the RDD Infos in `onBlockUpdate`, where the cached information is most update to date. I'm not sure if there is an ordering between `StageSubmitted` and `BlockUpdated` that can cause the cached information to be overridden.